### PR TITLE
Task 5: Import-Service 

### PR DESCRIPTION
**Pull request by SKolevatov**
https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task5-import-to-s3/task.md

- [X] 1 - File serverless.yml contains configuration for importProductsFile function

- [X] 3 - The importProductsFile lambda function returns a correct response which can be used to upload a file into the S3 bucket
- [X] 4 - Frontend application is integrated with importProductsFile lambda
- [X] 5 - The importFileParser lambda function is implemented and serverless.yml contains configuration for the lambda

- [X] +1 - async/await is used in lambda functions (In the current PR importFileParser works on callbacks, wasn't able to make it working using async/await, the implementation is in an alternative branch https://github.com/SKolevatov/nodejs-aws-be/tree/task5-1)
- [X] +1 - importProductsFile lambda is covered by unit tests (aws-sdk-mock can be used to mock S3 methods - https://www.npmjs.com/package/aws-sdk-mock)
- [X] +1 - At the end of the stream the lambda function should move the file from the uploaded folder into the parsed folder (move the file means that file should be copied into parsed folder, and then deleted from uploaded folder)

Link to FE PR: https://github.com/SKolevatov/nodejs-aws-fe/pull/4
Link to CloudFront website: https://d3q52y5xysba04.cloudfront.net/
To check the API, one may use https://driq9r5ej6.execute-api.eu-west-1.amazonaws.com/dev/import?name=test.csv

### CODE REVIEW

File: import-service/handlers/importProductsFile.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import { S3 } from 'aws-sdk';
+import 'source-map-support/register';
+
+const BUCKET = 'rss-task5';
+
+async function getSignedUrl(s3, params){
+    return new Promise((resolve,reject) => {
+        s3.getSignedUrl('putObject', params, (err, url) => {
```
Comment (Ilya-Valasiuk) [inline import-service/handlers/importProductsFile.ts]: aws already has method that return promise

```
await s3.getSignedUrlPromise('putObject', { ...params, Expires: 60 });
```

File: import-service/handlers/importFileParser.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            .on('error', err => console.error(err))
+            .pipe(csvParser())
+            .on('data', data => console.log(`DATA from ${key}:`, data))
+            .on('end', () => {
+                console.log(`${key} is read`);
+                s3.copyObject({
+                    Bucket: BUCKET,
+                    CopySource: `${BUCKET}/${key}`,
+                    Key: key.replace('uploaded', 'parsed')
+                }, (err) => {
```
Comment (Ilya-Valasiuk) [inline import-service/handlers/importFileParser.ts:30]: just interesting why we using callbacks here? Based on task we should use `async/await` in lambda functions

Comment (Ilya-Valasiuk) [reply import-service/handlers/importFileParser.ts:30]: I have checked PR description. this comment not relevant anymore.

File: import-service/package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  "name": "import-service",
+  "version": "1.0.0",
+  "description": "Serverless webpack example using Typescript",
+  "main": "handler.js",
+  "scripts": {
+    "test": "jest",
+    "deploy": "sls deploy"
+  },
+  "dependencies": {
+    "aws-sdk": "^2.791.0",
```
Comment (Ilya-Valasiuk) [inline import-service/package.json]: we can move sdk to `devDependencies ` so we will reduce bundle size

Comment (SKolevatov) [reply import-service/package.json]: Right, thanks!

File: import-service/handlers/importProductsFile.ts (hunk 1 above)
Comment (SKolevatov) [reply import-service/handlers/importProductsFile.ts]: Thanks, will use it.

**Pull request merged**